### PR TITLE
feat: in app tour

### PR DESCRIPTION
**Pull request by bryce-mcmath**
# Summary of Changes

Implementing our first in-app tour. The feature is optional and is not configured to be on by default. Here's what it looks like:
in_app_tour
Apologies for the many changes, it was a painful feature to implement. Also note that you ignore the double 'Welcome' text, I made the gif before merging in main with the latest changes.

# Related Issues

N/A

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [x] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [x] Updated documentation as needed for changed code and new or modified features;
- [x] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

File: packages/legacy/core/App/components/modals/AppGuideModal.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      color: ColorPallet.notification.infoText,
+    },
+    dismissIcon: {
+      alignSelf: 'center',
+    },
+  })
+
+  return (
+    <Modal transparent>
+      <TouchableOpacity onPress={onDismissPressed} accessible={false}>
```
Comment (jleach) [inline packages/legacy/core/App/components/modals/AppGuideModal.tsx]: Does this need `testID` or `accessabilityID` or are the child buttons good enough?

File: packages/legacy/core/App/components/tour/TourBox.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+/**
+ * A built-in TourBox component which can be used as a tooltip container for
+ * each step. While it's somewhat customizable, it's not required and can be
+ * replaced by your own component.
+ *
+ * @param props the component props
+ * @returns A TourBox React element
+ */
+export function TourBox(props: TourBoxProps): ReactElement {
+  const { leftText = 'Back', rightText = 'Next', title, hideLeft, hideRight, onLeft, onRight, children, stop } = props
```
Comment (jleach) [inline packages/legacy/core/App/components/tour/TourBox.tsx]: Translations for Back and Next?

File: packages/legacy/core/App/components/tour/TourOverlay.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+      setTooltipStyle({
+        left: tourMargin,
+        right: tourMargin,
+        top,
+      })
+    }
+  }, [spot.height, spot.width, spot.x, spot.y])
+
+  return (
+    <Modal animationType="fade" presentationStyle="overFullScreen" transparent={true} visible={current !== undefined}>
```
Comment (jleach) [inline packages/legacy/core/App/components/tour/TourOverlay.tsx:70]: Is `foo === undefined` ok in TS? I forget if its ok or we need `typeof foo !== 'undefined'`?

File: packages/legacy/core/App/screens/Tours.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+import { DispatchAction } from '../contexts/reducers/store'
+import { useStore } from '../contexts/store'
+import { useTheme } from '../contexts/theme'
+
+interface Option {
+  value: string
+  bool: boolean
+}
+
+const Tours = () => {
```
Comment (jleach) [inline packages/legacy/core/App/screens/Tours.tsx]: Bit of a different signature than we normally use: `const x = React.FC<XX> ...`.

File: packages/legacy/core/App/contexts/tour/tour-context.tsx [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+   * The spotlight layout.
+   */
+  spot: LayoutRectangle
+  /**
+   * The list of steps for the tour.
+   */
+  steps: TourStep[]
+}
+
+export const ZERO_SPOT: LayoutRectangle = {
```
Comment (jleach) [inline packages/legacy/core/App/contexts/tour/tour-context.tsx]: Origin maybe?

**Review by jleach — CHANGES_REQUESTED** [review]
A few comments for consideration. Will checkout and test short.

File: packages/legacy/core/App/components/tour/TourOverlay.tsx (hunk 3 above)
Comment (wadeking98) [reply packages/legacy/core/App/components/tour/TourOverlay.tsx:70]: I think `foo === undefined` works fine

**Review by wadeking98 — DISMISSED** [review]
once's jason's changes are addressed it looks good to me

**Review by jleach — APPROVED** [review]

**Pull request merged**